decentraland/unity-renderer
# Chore: hiccup when locking and unlocking the cursor

## What does this PR change?
Closes #4034 

This PR contains several performance improvements and clean-ups.

1. Removed remaining `Builder` assets and code parts (some where in `InputController`)
2. Converted LINQ `.Contrains` to `for`-loop in the `Update` of `InputController`, that slightly improved performance and removed garbage allocation.
3. Moved `Cursor` and `Helper` toast visibility change from `GameObject.SetActive()` to our `ShowHideAnimator` component
4. Added `isVisible` flag to several UI objects to prevent them Closing/Opening when they already in that state.


## How to test the changes?

1. Go to: https://play.decentraland.zone/?explorer-branch=chore/lock-unlock-cursor-hiccups
2. Verify that Input works as usual
3. Verify that locked cursor and the helper toast for it appearing as usual
4. Verify that Firends, Chat, VoiceChat, Passport UI's are closing and opening as usual (in all scenarios, also when locking curosr)

DEVS ONLY:
1. Checkout the branch and open with Deep profiling
2. Set DebugConfig to 100,100 and change `loki `t realm to `offline`
3. Start experience, wait a little bit, then start recording profiling data
4. Lock and unlock cursor several times
5. Move, Jump
6. Observe:

- ### `InputController` improved from `0.32 ms / 2.4 KB`  to `0.27 ms / 400 B`
![image](https://user-images.githubusercontent.com/35366872/216086397-a2fcf194-b658-44be-b205-623fd4521b4f.png)


- ### `MouseCatcher.CursorLock `improved from` 10.41 ms/ 4.3 KB `to `2.37 ms/ 1.7 KB` for the first call of locking the cursor
![image](https://user-images.githubusercontent.com/35366872/216088593-6b255885-41f4-4944-a621-31517480acba.png)

 - ### and from` 0.7 ms/ 2.3 KB` to `0.3 ms/ 0.8 KB` for all subsequent calls of the CursorLock
![image](https://user-images.githubusercontent.com/35366872/216089564-06da37a4-3fdb-411a-a2b6-186b3aee7814.png)


## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
